interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+        out['SRI'] = sri
+
+
+class Lindemann(object):
+    """The Lindemann falloff function."""
+    def get_yaml(self, out):
+        pass
+
+
+def convert(filename=None, outName=None, text=None):
+    # Reset global state, in case cti2yaml is being used as a module and convert
+    # is being called multiple times.
+    units('m', 'kmol', 'kg', 's', 'J/kmol', 'J', 'Pa')
+    standard_pressure(OneAtm)
+    global _motz_wise
+    _motz_wise = None
+    _elements.clear()
+    _species.clear()
+    _speciesnames.clear()
+    _phases.clear()
+    _reactions.clear()
+    _reactions['reactions'] = []
+
+    if filename is not None:
+        filename = os.path.expanduser(filename)
+        base = os.path.basename(filename)
+        root, _ = os.path.splitext(base)
+        dataset(root)
+    if outName is None and _name != 'noname':
+        outName = _name + '.yaml'
+
+    open_kw = {'encoding': 'latin-1'} if sys.version_info.major == 3 else {}
+    open_mode = 'r' if sys.version_info.major == 3 else 'rU'
+    try:
+        if filename is not None:
+            with open(filename, open_mode, **open_kw) as f:
+                code = compile(f.read(), filename, 'exec')
+        else:
+            code = compile(text, '<string>', 'exec')
+        exec(code)
+    except SyntaxError as err:
+        # Show more context than the default SyntaxError message
+        # to help see problems in multi-line statements
+        if filename:
+            text = open(filename, open_mode).readlines()
+        else:
+            text = text.split('\n')
+        _printerr('%s in "%s" on line %i:\n' % (err.__class__.__name__,
+                                                err.filename,
+                                                err.lineno))
+        _printerr('|  Line |')
+        for i in range(max(err.lineno-6, 0),
+                       min(err.lineno+3, len(text))):
+            _printerr('| % 5i |' % (i+1), text[i].rstrip())
+            if i == err.lineno-1:
+                _printerr(' '* (err.offset+9) + '^')
+        _printerr()
+        sys.exit(3)
+    except Exception as err:
+        import traceback
+
+        if filename:
+            text = open(filename, open_mode).readlines()
+        else:
+            text = text.split('\n')
+            filename = '<string>'
+        tb = traceback.extract_tb(sys.exc_info()[2])
+        lineno = tb[-1][1]
+        if tb[-1][0] == filename:
+            # Error in input file
+            _printerr('%s on line %i of %s:' % (err.__class__.__name__, lineno, filename))
+            _printerr(err)
+            _printerr('\n| Line |')
+
+            for i in range(max(lineno-6, 0),
+                           min(lineno+3, len(text))):
+                if i == lineno-1:
+                    _printerr('> % 4i >' % (i+1), text[i].rstrip())
+                else:
+                    _printerr('| % 4i |' % (i+1), text[i].rstrip())
+        else:
+            # Error in ctml_writer or elsewhere
```

Comment:
Should this be something other than `ctml_writer`?

Reply:
🤦‍♂